lib/http-incoming.js

```diff
@@ -15,9 +14,24 @@ const _url = Symbol('podium:httpincoming:url');
 const _css = Symbol('podium:httpincoming:css');
 const _js = Symbol('podium:httpincoming:js');
 
+const protocolFromRequest = (request = {}) => {
+    const secure =
+        request.secure || (request.connection && request.connection.encrypted);
+    return secure ? 'https:' : 'http:';
+};
+
+const urlFromRequest = (request, protocol) => {
+    try {
+        return new URL(request.url, `${protocol}//${request.headers.host}`);
+    } catch (err) {
+        return {};
+    }
+};
+
 const PodiumHttpIncoming = class PodiumHttpIncoming {
     constructor(request = {}, response = {}, params = {}) {
-        const url = originalUrl(request);
+        const protocol = protocolFromRequest(request);
+        const url = urlFromRequest(request, protocol);
```

Comment:
could maybe use some additional tests to verify that this is working as expected